### ISSUE TITLE
Map accessor fails in some specific circumstances (static, auto initialised, non-stored)

### ISSUE THREAD

**Issue by DiasFranciscoA** [YoYoGames/GameMaker-Bugs#6472]
### Description

This will produce a runtime error with the message:

unable to convert string "<key>" to int64 (if key is a string)
trying to index a variable which is not an array (if the key is a number)

[bug_with_map_accessor.zip](https://github.com/user-attachments/files/16067486/bug_with_map_accessor.zip)

```
new function C() constructor {

	static __ = {};
	with (__) {
		variable = ds_map_create();
	}

	static access_variable = function(_name) {
		return __.variable[? _name];
	}
}();

show_debug_message(C.access_variable("greet"));
show_debug_message(C.access_variable(12));
```

### Steps To Reproduce

1. Start GameMaker
2. Use demo to see the issue (Script1)
3. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.4.1 (Monthly), 2024.600 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by rwkay** [comment]
Fixed in 2024.8

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.800.0.591  Runtime v2024.800.0.615
